Documentation/platform/troubleshooting.md

```diff
@@ -90,6 +90,27 @@ kubectl -n monitoring port-forward svc/prometheus-operated 9090:9090
 
 If the command runs successfully, you should be able to access the [Prometheus server UI](http://localhost:9090/) via localhost. From there you can check the live configuration and the discovered targets.
 
+### Debugging why monitoring resource spec changes are not reconciled
+
+The Prometheus Operator will reject invalid resources and not reconcile them in the Prometheus configuration, for instance if the `ServiceMonitor` object has a references to a secret which doesn't exist in the cluster. When it happens the Operator emits a Kubernetes Event detailing the issue.
```

Comment:
Note that it also applies to AlertmanagerConfig resources so we might want a more generic statement.

Reply:
Actually dropped the example to make is clearer as `invalid resources` with the list below is descriptive enough.